lib/vagrant-openshift/action/push_openshift_images.rb

```diff
@@ -37,50 +37,44 @@ def fix_insecure_registry_cmd(registry_url)
           %{
 set -e
+pushd /data/src/github/openshift/#{image_name}
+git checkout #{git_ref}
+git_ref=$(git rev-parse --short HEAD)
+echo "Pushing image #{image_name}:$git_ref..."
+docker tag -f #{image_name}-centos7 #{registry}#{image_name}-centos7:$git_ref
+docker tag -f #{image_name}-centos7 #{registry}#{image_name}-centos7:latest
+docker tag -f #{image_name}-centos7 docker.io/#{image_name}-centos7:latest
+docker tag -f #{image_name}-rhel7 #{registry}#{image_name}-rhel7:$git_ref
+docker tag -f #{image_name}-rhel7 #{registry}#{image_name}-rhel7:latest
+# FIXME: Paralelize this
+docker push -f #{registry}#{image_name}-centos7:$git_ref
+docker push -f #{registry}#{image_name}-centos7:latest
+docker push -f docker.io/#{image_name}-centos7:latest
+docker push -f #{registry}#{image_name}-rhel7:$git_ref
+docker push -f #{registry}#{image_name}-rhel7:latest
+popd
+          }
+        end
 
-# so we can call sti
-PATH=/data/src/github.com/openshift/source-to-image/_output/go/bin:/data/src/github.com/openshift/source-to-image/_output/local/go/bin:$PATH
-
+        def build_image(image_name, git_ref, repo_url, registry)
+          %{
 dest_dir="/data/src/github/openshift/#{image_name}"
 rm -rf ${dest_dir}; mkdir -p ${dest_dir}
-if git clone #{repo_url} ${dest_dir}; then
-  cd ${dest_dir}
-  git checkout #{git_ref}
-  git_ref=$(git rev-parse --short HEAD)
-  echo "Building and testing #{image_name}:$git_ref"
-  if ! make test TARGET=centos7; then
-    echo "ERROR: #{image_name}-centos7 failed testing."
-    exit 1
-  fi
-  if make build TARGET=centos7; then
-    docker tag -f #{image_name}-centos7 #{registry}#{image_name}-centos7:$git_ref
-    docker tag -f #{image_name}-centos7 #{registry}#{image_name}-centos7:latest
-    docker tag -f #{image_name}-centos7 docker.io/#{image_name}-centos7:latest
-    docker push -f #{registry}#{image_name}-centos7:$git_ref
-    docker push -f #{registry}#{image_name}-centos7:latest
-    docker push -f docker.io/#{image_name}-centos7:latest
-  else
-    echo "ERROR: Failed to build #{image_name}-centos7"
-    exit 1
-  fi
-
-  if ! make test TARGET=rhel7; then
-    echo "ERROR: #{image_name}-rhel7 failed testing."
-    exit 1
-  fi
-  if make build TARGET=rhel7; then
-    docker tag -f #{image_name}-rhel7 #{registry}#{image_name}-rhel7:$git_ref
-    docker tag -f #{image_name}-rhel7 #{registry}#{image_name}-rhel7:latest
-    docker push -f #{registry}#{image_name}-rhel7:$git_ref
-    docker push -f #{registry}#{image_name}-rhel7:latest
-  else
-    echo "ERROR: Failed to build #{image_name}-rhel7"
-    exit 1
-  fi
-fi
+set -e
+pushd ${dest_dir}
+git init && git remote add -t master origin #{repo_url}
+git fetch && git checkout #{git_ref}
+git_ref=$(git rev-parse --short HEAD)
+echo "Building and testing #{image_name}-centos7:$git_ref ..."
+make test TARGET=centos7
+make build TARGET=centos7
+echo "Building and testing #{image_name}-rhel7:$git_ref ..."
+make test TARGET=rhel7
+make build TARGET=rhel7
+popd
```

Comment:
sure, but if no error occurs you've left it in set -e mode for the next block.

I think each block should be setting up -e/+e as it desires, otherwise you've got unexpected behavior when the next block runs, right?